File: app/src/main/java/se/hactar/movieregister/ui/MovieListFragment.java
package se.hactar.movieregister.ui;


import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import se.hactar.movieregister.R;
import se.hactar.movieregister.helper.DebugLifcycleObserver;
import se.hactar.movieregister.viewmodel.MovieListViewModel;

public class MovieListFragment extends Fragment {

    private RecyclerView recyclerView;

    @Nullable
    @Override
    public View onCreateView(final LayoutInflater inflater, final ViewGroup container, final Bundle savedInstanceState) {
        getLifecycle().addObserver(new DebugLifcycleObserver(this));
        View view = inflater.inflate(R.layout.fragment_movie_list, container, false);
        recyclerView = view.findViewById(R.id.movies);
        MovieAdapter adapter = new MovieAdapter();
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        MovieListViewModel model = ViewModelProviders.of(getActivity()).get(MovieListViewModel.class);
        model.getLiveDataMovies().observe(this, adapter::addAll);
        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        recyclerView.setAdapter(null);
    }
}

File: app/src/main/java/se/hactar/movieregister/ui/MovieListFragment.kt
package se.hactar.movieregister.ui


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import se.hactar.movieregister.R
import se.hactar.movieregister.helper.DebugLifecycleObserver
import se.hactar.movieregister.viewmodel.MovieListViewModel

class MovieListFragment : Fragment() {

    private lateinit var recyclerView: RecyclerView

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        lifecycle.addObserver(DebugLifecycleObserver(this))
        val view = inflater.inflate(R.layout.fragment_movie_list, container, false)
        recyclerView = view.findViewById(R.id.movies)
        val adapter = MovieAdapter()
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(activity)
        val model = ViewModelProvider(requireActivity()).get(MovieListViewModel::class.java)
        model.liveDataMovies().observe(viewLifecycleOwner, { adapter.addAll(it!!) })
        return   view
    }

    override fun onDestroyView() {
        super.onDestroyView()
        recyclerView.adapter = null
    }
}
